lib/bundle/brew_installer.rb

```diff
@@ -92,6 +92,9 @@ def service_change_state!(verbose:)
 
     def link_change_state!(verbose: false)
       case @link
+      when :overwrite
+        puts "Overwriting links for #{@name} formula." if verbose
+        Bundle.system(HOMEBREW_BREW_FILE, "link", "--force", "--overwrite", @name, verbose: verbose)
```

Comment:
Are you sure `--force` is needed here? That should only be needed if it's keg-only.

Reply:
Right - it's meant to support overwriting links with a keg-only formula. Given that `--force` is effectively a no-op for non-key-only formulae, I figured it's clearer to use a single invocation that works for any formula.

Reply:
@jeremy I think `--force` should only be passed when it's keg-only, we have that information already in memory and it feels like a cleaner implementation.

Reply:
That state ("keg-only and unlinked") is not presently available, but could be with changes to BrewDumper.

BrewDumper.formulae sets `link?` to one of three values which capture four distinct states:
| dumped | logic | keg_only | linked |
|---|---|---|---|
| `link?: true` | keg_only? && linked? | true | true |
| `link?: false` | !keg_only? && !linked? | false | false |
| `link?: nil` | keg_only? && !linked? | true | false |
| `link?: nil` | !keg_only? && linked? | false | true |

We could have BrewDumper use `link?: :overwrite` to capture the `keg_only? && !linked?` state, or we could split the compound `link?` state into `linked?` and `keg_only?` booleans.

Reply:
> We could have BrewDumper use link?: :overwrite to capture the keg_only? && !linked? state, or we could split the compound link? state into linked? and keg_only? booleans.

Fine with either option 👍🏻